pypeit/scripts/ql_vlt_fors2.py

```diff
@@ -0,0 +1,465 @@
+        parset = par.PypeItPar.from_cfg_lines(cfg_lines=spectrograph_cfg_lines,
+                                              merge_with=config_lines(args))
+        science_path = os.path.join(parset['rdx']['redux_path'], parset['rdx']['scidir'])
+
+        target = spectrograph.get_meta_value(files[0], 'target')
+        mjds = np.zeros(nfiles)
+        for ifile, file in enumerate(files):
+            mjds[ifile] = spectrograph.get_meta_value(file, 'mjd', ignore_bad_header=True,
+                                                      no_fussing=True)
+        files = files[np.argsort(mjds)]
+
+        # Calibration Master directory
+        #TODO hardwired for now
+        master_dir ='./'
+        #master_dir = resource_filename('pypeit', 'data/QL_MASTERS') \
+        #    if args.master_dir is None else args.master_dir
+        if not os.path.isdir(master_dir):
+            msgs.error(f'{master_dir} does not exist!  You must install the QL_MASTERS '
+                       'directory; download the data from the PypeIt dev-suite Google Drive and '
+                       'either define a QL_MASTERS environmental variable or use the '
+                       'pypeit_install_ql_masters script.')
+
+        # Define some hard wired master files here to be later parsed out of the directory
+        fors2_grism = spectrograph.get_meta_value(files[0], 'dispname')
+        fors2_masters = os.path.join(master_dir, 'FORS2_MASTERS', fors2_grism)
+
+
+        bias_masterframe_name = \
+            utils.find_single_file(os.path.join(fors2_masters, "MasterBias*"))
+        slit_masterframe_name \
+            = utils.find_single_file(os.path.join(fors2_masters, "MasterSlits*"))
+        tilts_masterframe_name \
+            = utils.find_single_file(os.path.join(fors2_masters, "MasterTilts*"))
+        wvcalib_masterframe_name \
+            = utils.find_single_file(os.path.join(fors2_masters, 'MasterWaveCalib*'))
+        std_spec1d_file = utils.find_single_file(os.path.join(fors2_masters, 'spec1d_*'))
+        sensfunc_masterframe_name = utils.find_single_file(os.path.join(fors2_masters, 'sens_*'))
+
+        # TODO make and impelement sensfunc
+        if (bias_masterframe_name is None or not os.path.isfile(bias_masterframe_name)) or \
+                (slit_masterframe_name is None or not os.path.isfile(slit_masterframe_name)) or \
+                (tilts_masterframe_name is None or not os.path.isfile(tilts_masterframe_name)) or \
+                (std_spec1d_file is None or not os.path.isfile(std_spec1d_file)):
+            # or (sensfunc_masterframe_name is None or not os.path.isfile(sensfunc_masterframe_name)):
+            msgs.error('Master frames not found.  Check that environment variable QL_MASTERS '
+                       'points at the Master Calibs')
+
+        # We need the platescale
+
+        # Get detector (there's only one)
+        #det = 1 # MOSFIRE has a single detector
```

Comment:
Doesn't FORS2 have 2 detectors? Does any of the code below need to get changed to include the second detector?

Reply:
Deprecated.